File: server/app/schemas/team.py
from pydantic import BaseModel


class Team(BaseModel):
    id: int
    full_name: str
    abbreviation: str
    nickname: str
    city: str
    state: str
    year_founded: int
    arena: str | None = None
    head_coach: str | None = None
    general_manager: str | None = None

File: client/src/types/Team.ts
export interface Team {
    id: number;
    full_name: string;
    abbreviation: string;
    nickname: string;
    city: string;
    state: string;
    year_founded: number;
    arena: string | null;
    head_coach: string | null;
    general_manager: string | null;
}
